### PR TITLE
(WIP) Added command to discover AWS IAM Profile and AWS S3 Bucket mapping

### PR DESCRIPTION
**Pull request by nfx**
This pull request introduces a new method `iam_profiles()` in the file `aws.py`. This method is a Work In Progress (WIP) and lists all AWS IAM roles, their attached and inline policies, and S3 bucket permissions for each role. It uses the `subprocess` module to run AWS CLI commands and parse their output to extract the necessary information.

There is also a new command `aws-iam-profiles` added to the `labs.yml` file, which corresponds to the new `aws_iam_profiles()` method.

Regarding the tests, the `test_aws.py` file has been updated with new scenarios to test the `iam_profiles()` method. The tests include:

1. Testing successful execution of the `iam_profiles()` method and checking the output for correctness.
2. Testing the method when there are no IAM roles available, and checking the error message.
3. Testing the method when the AWS CLI is not configured, and checking the error message.

In summary, this pull request adds a new method to list AWS IAM roles, their policies, and S3 bucket permissions, along with new test scenarios to validate the method's functionality.

### CODE REVIEW

**Review by dmoore247 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/assessment/aws.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+
+def iam_profiles():
```
[the remark this answers is not in the document]
Comment (nfx) [reply src/databricks/labs/ucx/assessment/aws.py:2]: i don't think so

File: src/databricks/labs/ucx/assessment/aws.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    import subprocess
+    import json
+
+    def run_command(command):
+        process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```
[the remark this answers is not in the document]
Comment (nfx) [reply src/databricks/labs/ucx/assessment/aws.py:7]: > Why not use the awscli python package?

because it's not pre-approved by legal.

File: src/databricks/labs/ucx/assessment/aws.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nfx) [reply src/databricks/labs/ucx/assessment/aws.py:2]: yeap, it's still a WIP
